### PR TITLE
OpenSSL compatibility fix

### LINKED ISSUE #167: OpenSSL changed format of x509 outputs
Default output format has changed between OpenSSL versions 1.0.2 and 1.1.1. The `-nameopt` argument can be used, but it seems to have inconsistent results if you use the `-nameopt compat` format. There is the option of "negated" format options, e.g `-nameopt -online`, but can't find a combo that helps. This implies we'd need to check what version of OpenSSL is being used, or change the format of the subject strings everywhere. 

1.0.2:
```
$ openssl x509 -noout -subject < /etc/grid-security/certificates/UKeScienceCA-2B.pem
subject= /C=UK/O=eScienceCA/OU=Authority/CN=UK e-Science CA 2B

$ openssl x509 -noout -nameopt compat -subject < /etc/grid-security/certificates/UKeScienceCA-2B.pem
subject= C=UK, O=eScienceCA, OU=Authority, CN=UK e-Science CA 2B

$ openssl x509 -noout -nameopt oneline -subject < /etc/grid-security/certificates/UKeScienceCA-2B.pem
subject= C = UK, O = eScienceCA, OU = Authority, CN = UK e-Science CA 2B

$ openssl x509 -noout -nameopt -oneline -subject < /etc/grid-security/certificates/UKeScienceCA-2B.pem
subject= C=UK, O=eScienceCA, OU=Authority, CN=UK e-Science CA 2B

```
1.1.1:
```
$ openssl x509 -noout -subject < /etc/grid-security/certificates/UKeScienceCA-2B.pem
subject=C = UK, O = eScienceCA, OU = Authority, CN = UK e-Science CA 2B

$ openssl x509 -noout -nameopt compat -subject < certificate.crt
subject=/C=UK/L=Didcot/O=RAL/CN=this

$ openssl x509 -noout -nameopt oneline -subject < /etc/grid-security/certificates/UKeScienceCA-2B.pem
subject=C = UK, O = eScienceCA, OU = Authority, CN = UK e-Science CA 2B

$ openssl x509 -noout -nameopt -oneline -subject < /etc/grid-security/certificates/UKeScienceCA-2B.pem
subject=/C=UK/O=eScienceCA/OU=Authority/CN=UK e-Science CA 2B
```

### PR DESCRIPTION
**Pull request by RedProkofiev**
Resolves #167 

At @jrha 's suggestion, to solve a formatting conflict, this PR adds pyOpenSSL as a dependency and uses it to handle the distinguished name formatting.  pyOpenSSL isn't currently used for anything else.

### CODE REVIEW

**Review by gregcorbett — DISMISSED** [review]
need to check if this still supports python 2.7

**Comment by RedProkofiev** [comment]
> need to check if this still supports python 2.7

it does not lol, pyopenssl is for 3+
if this milestone is meant to be for python 2 and 3 this change needs to be stashed for a py3 only build

**Comment by RedProkofiev** [comment]
Confirmed, works with Python 2

File: ssm/crypto.py [hunk 1]
```diff
@@ -297,17 +298,23 @@ def verify_cert_path(certpath, capath, check_crls=True):
...
+        subject_x509name = OpenSSL.crypto.load_certificate(
+            type=OpenSSL.crypto.FILETYPE_PEM,
+            buffer=certstring
+            ).get_subject()
+    except Exception as error:
         log.error(error)
         raise CryptoException('Failed to get subject: %s' % error)
 
-    subject = subject.strip()[9:]  # remove 'subject= ' from the front
+    subject = "".join("/{:s}={:s}".format(name.decode(), value.decode())
```
[the remark this answers is not in the document]
Comment (RedProkofiev) [reply ssm/crypto.py:317]: Added

**Comment by tofu-rocketry** [comment]
Looks like this needs a rebase.

File: ssm/crypto.py [hunk 2]
```diff
@@ -297,17 +298,25 @@ def verify_cert_path(certpath, capath, check_crls=True):
...
+            buffer=certstring
+            ).get_subject()
+    except Exception as error:
         log.error(error)
-        raise CryptoException('Failed to get subject: %s' % error)
+        log.error(CryptoException)
+        raise error
+
+    # Outputs DN in slash-separated format
+    subject = "".join("/{:s}={:s}".format(name.decode(), value.decode())
```
Comment (tofu-rocketry) [inline ssm/crypto.py:317]: Could you set this up as a function? As it might end up useful elsewhere and means we can unit test it.

**Review by tofu-rocketry — CHANGES_REQUESTED** [review]

File: ssm/crypto.py (hunk 2 above)
Comment (RedProkofiev) [reply ssm/crypto.py]: Most recent commit has moved this to a function.

**Comment by tofu-rocketry** [comment]
Do you think you could add a unit test for the new function?

**Pull request merged**